### PR TITLE
Common: Fix handlers that are not properly triggered.

### PR DESCRIPTION
**Pull request by guits**
Until now, only the first task were executed.
The idea here is to use `listen` statement to be able to notify multiple
handler and regroup all of them in `./handlers/main.yml` as notifying an
included handler task is not possible.

Signed-off-by: Guillaume Abrioux <gabrioux@redhat.com>

### CODE REVIEW

**Comment by guits** [comment]
jenkins test kraken-ansible2.2-docker_dedicated_journal

**Comment by guits** [comment]
jenkins test kraken-ansible2.2-journal_collocation

File: roles/ceph-common/templates/restart_mons_daemons.sh.j2 [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+DELAY="{{ handler_health_mon_check_delay }}"
+MONITOR_NAME="{{ monitor_name }}"
+CLUSTER="{{ cluster }}"
+
+systemctl restart ceph-mon@$MONITOR_NAME
+sleep 2
+stat /var/run/ceph/${CLUSTER}-mon.${MONITOR_NAME}.asok || exit 1
+
+while [ $RETRIES -ne 0 ]; do
+	MEMBERS=$(ceph --cluster ${CLUSTER} -s --format json | sed -r 's/.*"quorum_names":(\[[^]]+\]).*/\1/')
```
Comment (leseb) [inline roles/ceph-common/templates/restart_mons_daemons.sh.j2]: wrong indent, use an indent of 2

File: roles/ceph-common/templates/restart_mons_daemons.sh.j2 [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+RETRIES="{{ handler_health_mon_check_retries }}"
+DELAY="{{ handler_health_mon_check_delay }}"
+MONITOR_NAME="{{ monitor_name }}"
+CLUSTER="{{ cluster }}"
+
+systemctl restart ceph-mon@$MONITOR_NAME
+sleep 2
```
Comment (leseb) [inline roles/ceph-common/templates/restart_mons_daemons.sh.j2]: please add a comment to explain why

File: roles/ceph-common/templates/restart_osds_daemons.sh.j2 [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+RETRIES="{{ handler_health_osd_check_retries }}"
+DELAY="{{ handler_health_osd_check_delay }}"
+CEPH_CLI="--name client.bootstrap-osd --keyring /var/lib/ceph/bootstrap-osd/{{ cluster }}.keyring --cluster {{ cluster }}"
+
+for id in $(ls /var/lib/ceph/osd/ | sed 's/.*-//'); do
+	systemctl restart ceph-osd@$id
```
Comment (leseb) [inline roles/ceph-common/templates/restart_osds_daemons.sh.j2]: wrong indent, use an indent of 2

File: roles/ceph-common/templates/restart_osds_daemons.sh.j2 [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+	sleep 2
```
Comment (leseb) [inline roles/ceph-common/templates/restart_osds_daemons.sh.j2]: ditto

File: roles/ceph-common/templates/restart_osds_daemons.sh.j2 [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+	stat /var/run/ceph/{{ cluster }}-osd.$id.asok || exit 1
+	while [ $RETRIES -ne 0 ]; do
+		ceph $CEPH_CLI -s | grep -sq 'active+clean'
```
Comment (leseb) [inline roles/ceph-common/templates/restart_osds_daemons.sh.j2]: wrong indent, use an indent of 2

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-docker_cluster

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-update_cluster

File: roles/ceph-common/handlers/main.yml [hunk 6]
```diff
@@ -4,17 +4,67 @@
...
+      src: restart_mons_daemons.sh.j2
+      dest: /tmp/restart_mons_daemons.sh
+      owner: root
+      group: root
+      mode: 0750
+    listen: "restart ceph mons"
 
-- name: restart ceph osds
-  include: "./restart-osd.yml"
+  - name: restart ceph mons daemon(s)
```
Comment (leseb) [inline roles/ceph-common/handlers/main.yml]: s/mons/mon/

File: roles/ceph-common/handlers/main.yml [hunk 7]
```diff
@@ -4,17 +4,67 @@
     update-cache: yes
   when: ansible_os_family == 'Debian'
 
-- name: restart ceph mons
-  include: "./restart-mon.yml"
+- block:
+  - name: copy mon restart script
+    template:
+      src: restart_mons_daemons.sh.j2
```
Comment (leseb) [inline roles/ceph-common/handlers/main.yml]: s/restart_mons_daemons/restart_mon_daemon/

File: roles/ceph-common/handlers/main.yml [hunk 8]
```diff
@@ -4,17 +4,67 @@
...
+      dest: /tmp/restart_mons_daemons.sh
+      owner: root
+      group: root
+      mode: 0750
+    listen: "restart ceph mons"
 
-- name: restart ceph osds
-  include: "./restart-osd.yml"
+  - name: restart ceph mons daemon(s)
+    command: /tmp/restart_mons_daemons.sh
```
Comment (leseb) [inline roles/ceph-common/handlers/main.yml]: s/restart_mons_daemons/restart_mon_daemon/

File: roles/ceph-common/handlers/main.yml [hunk 9]
```diff
@@ -4,17 +4,67 @@
...
+  when:
+    - mon_group_name in group_names
+
+# This does not just restart OSDs but everything else too. Unfortunately
+# at this time the ansible role does not have an OSD id list to use
+# for restarting them specifically.
+- block:
+  - name: copy osd restart script
+    template:
+      src: restart_osds_daemons.sh.j2
```
Comment (leseb) [inline roles/ceph-common/handlers/main.yml]: s/restart_osds_daemons/restart_osd_daemon/

File: roles/ceph-common/handlers/main.yml [hunk 10]
```diff
@@ -4,17 +4,67 @@
...
+      dest: /tmp/restart_osds_daemons.sh
```
Comment (leseb) [inline roles/ceph-common/handlers/main.yml]: s/restart_osds_daemons/restart_osd_daemon/

File: roles/ceph-common/handlers/main.yml [hunk 11]
```diff
@@ -4,17 +4,67 @@
...
+      owner: root
+      group: root
+      mode: 0750
+    listen: "restart ceph osds"
+
+  - name: restart ceph osds daemon(s)
+    command: /tmp/restart_osds_daemons.sh
```
Comment (leseb) [inline roles/ceph-common/handlers/main.yml]: s/restart_osds_daemons/restart_osd_daemon/

File: roles/ceph-common/templates/restart_mons_daemons.sh.j2 [hunk 12]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+RETRIES="{{ handler_health_mon_check_retries }}"
+DELAY="{{ handler_health_mon_check_delay }}"
+MONITOR_NAME="{{ monitor_name }}"
+CLUSTER="{{ cluster }}"
+
+systemctl restart ceph-mon@${MONITOR_NAME}
+# We need to wait because it may take some time for the socket to actually exists
+sleep 2
```
Comment (leseb) [inline roles/ceph-common/templates/restart_mons_daemons.sh.j2]: this is really prone to error, please do a loop and wait for 10 secs, then exit 1 with a message.

File: roles/ceph-common/templates/restart_osds_daemons.sh.j2 [hunk 13]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+RETRIES="{{ handler_health_osd_check_retries }}"
+DELAY="{{ handler_health_osd_check_delay }}"
+CEPH_CLI="--name client.bootstrap-osd --keyring /var/lib/ceph/bootstrap-osd/{{ cluster }}.keyring --cluster {{ cluster }}"
+
+for id in $(ls /var/lib/ceph/osd/ | sed 's/.*-//'); do
+  systemctl restart ceph-osd@${id}
+  # We need to wait because it may take some time for the socket to actually exists
+  sleep 2
```
Comment (leseb) [inline roles/ceph-common/templates/restart_osds_daemons.sh.j2]: ditto

**Comment by guits** [comment]
jenkins test kraken-ansible2.2-docker_dmcrypt_journal_collocation

**Pull request merged**